lib/course_planner/teachers.ex

```diff
@@ -48,4 +48,14 @@ defmodule CoursePlanner.Teachers do
   defp format_error({:ok, teacher}, _), do: {:ok, teacher}
   defp format_error({:error, changeset}, teacher), do: {:error, teacher, changeset}
 
+  def courses(teacher_id) do
+    Repo.all(from oc in OfferedCourse,
+      left_join: oct in "offered_courses_teachers", on: oct.offered_course_id == oc.id,
+      left_join: u in User, on: u.id == oct.teacher_id,
```

Comment:
We only use the user id from the user, so we can remove this join and replace the condition
```elixir
where: u.id == ^teacher_id
```
by 
```elixir
where: oct.teacher_id == ^teacher_id
```